core/src/main/java/io/grpc/Contexts.java

```diff
@@ -0,0 +1,239 @@
+package io.grpc;
+
+/**
+ * Utility methods for working with {@link Context}s in GRPC.
+ */
+public class Contexts {
+
+  /**
+   * Create a {@link io.grpc.ServerInterceptor} that will apply a transformation to the
+   * context associated with each {@link ServerCall} and ensure that it is
+   * {@link io.grpc.Context#current()} when dispatching calls into application callbacks.
+   * @param transformer that applies a transformation to the context.
+   * @return a {@link io.grpc.ServerInterceptor} the applies the transformation that can be bound
+   *         to a {@link io.grpc.ServerServiceDefinition}.
+   */
+  public static ServerInterceptor transformAsServerInterceptor(
```

Comment:
@lukesandberg this probably does what you want in terms of the ability to augment context on incoming calls.